packages/discovery-provider/src/queries/query_helpers.py

```diff
@@ -557,7 +558,27 @@ def populate_track_metadata(
 
 
 def _populate_gated_track_metadata(session, tracks, current_user_id):
-    if not tracks or not current_user_id:
+    if not tracks:
+        return
+    if not current_user_id:
+        for track in tracks:
+            stream_conditions = track.get("stream_conditions")
```

Comment:
this feels like a bit of a strange place to do this kind of check. why can't clients just know what to serve without doing this kind of data wrangling?

Reply:
they can! but i wanted the client to just simply use the `track.access` field, without being smart

Reply:
clients need this, to be able to know what to serve. otherwise, they would do some null checking, or logged in check. this is much cleaner imo

Reply:
it's cleaner on the API side, I agree, but it's just fragile code that will have bugs down the line almost certainly. Not blocking merge though. We should think more about the architecture here though I think